src/plugins/intel_gna/src/transformations/insert_identity_layer.cpp

```diff
@@ -231,3 +232,98 @@ bool IdentityCandidatesCleanup::run_on_model(const std::shared_ptr<ov::Model>& f
+    });
+}
+
+size_t InsertIdentityForPrecAgnosticConcatInput::find_prev_layer_output_index(shared_ptr<ov::Node> prev,
+                                                                              shared_ptr<ov::Node> next) {
+    for (const auto& output : prev->outputs()) {
+        for (const auto& input : output.get_target_inputs()) {
+            if (input.get_node() == next.get()) {
+                return output.get_index();
+            }
+        }
+    }
+    return 0;
+}
+
+bool InsertIdentityForPrecAgnosticConcatInput::insert_identity_after_nodes(const vector<shared_ptr<ov::Node>>& nodes,
+                                                                           const shared_ptr<ov::Node>& next) {
+    bool is_graph_modified = false;
+    for (auto node : nodes) {
+        size_t index = find_prev_layer_output_index(node, next);
+        insert_identity_layer_after(node, index);
+        is_graph_modified = true;
+    }
+    return is_graph_modified;
+}
+
+bool InsertIdentityForPrecAgnosticConcatInput::has_fq_on_any_input(const shared_ptr<ov::Node> concat_node) {
+    auto is_not_fq = [](shared_ptr<ov::Node> node) -> bool {
+        return !is_parameter(node) && !is_constant(node) && !is_fake_quantize(node);
+    };
+    for (size_t i = 0; i < concat_node->get_input_size(); i++) {
+        auto concat_input_node = concat_node->get_input_node_shared_ptr(i);
+        auto prev_node = get_prev_node_skipping_certain(concat_input_node, is_not_fq);
+        if (!is_parameter(prev_node) && !is_constant(prev_node)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+vector<shared_ptr<ov::Node>> InsertIdentityForPrecAgnosticConcatInput::get_nodes_for_identity_insertion(
+    const shared_ptr<ov::Node>& concat_node) {
+    auto not_able_to_hold_sf = [](shared_ptr<ov::Node> node) -> bool {
+        return is_gna_precision_agnostic(node) || is_fake_quantize(node) || is_read_value(node);
+    };
+
+    vector<shared_ptr<ov::Node>> nodes;
+    for (size_t i = 0; i < concat_node->get_input_size(); i++) {
+        auto concat_input_node = concat_node->get_input_node_shared_ptr(i);
+        auto prev_node = get_prev_node_skipping_certain(concat_input_node, not_able_to_hold_sf);
+        if ((is_parameter(prev_node) || is_constant(prev_node)) && (!is_parameter(concat_input_node))) {
+            nodes.emplace_back(concat_input_node);
+        }
+    }
+    return nodes;
+}
+
+bool InsertIdentityForPrecAgnosticConcatInput::insert_identity_for_prec_agnostic_concat_inputs(
+    const shared_ptr<ov::Node>& concat_node) {
+    if (all_inputs_point_the_same_node(concat_node)) {
+        return false;
+    }
+
+    if (!has_fq_on_any_input(concat_node)) {
+        return false;
+    }
+
+    const auto& nodes = get_nodes_for_identity_insertion(concat_node);
+
+    // Skip adding Identity if none of inputs can hold scale factors
+    bool no_input_can_hold_sf = concat_node->get_input_size() == nodes.size();
```

Comment:
based on comment shouldn't condition look like `if( nodes.empty())` ?

Reply:
No, because `nodes` contain all inputs, which are not able to hold scale factors.